main.go

```diff
@@ -151,6 +155,46 @@ func updateNDK(revision string) error {
 	return nil
 }
 
+func unWrapNDKDirIfNeeded(ndkHome string, revision string) error {
```

Comment:
Instead of using strict/hardcoded pattern to find the corresponding directory, please try to use a more free variant of finding the directory which contains the platform directory.

One of a solution can be to use https://golang.org/pkg/path/filepath/#Glob and don't forget that you should use absolute paths in Glob. For example: to run a glob for a given set of globs, in priority, for example 0: `/currently/given/path/platforms` 1: `/currently/given/path/*/platforms` so this way the depth of the walk is managed.

The other solution would be to have a filepath.Walk that exits on the first occurence if the currently found fileinfo is a directory and its name is platforms. The depth of the filepath recursion in this case is unmanaged here.

Also please do not move files/dirs but set the requirements to use the new path instead.

So, you can have:
- a method that ensures the given ndk path, which means that it will return a valid ndk path
- set this path with os.Setenv(current process) and envman(following processes) as well.
- move forward as usually

Reply:
Updated to walk.